mls-rs/src/group/ciphertext_processor.rs

```diff
@@ -298,19 +315,24 @@ mod test {
     }
 
     #[cfg_attr(not(mls_build_async), maybe_async::must_be_sync)]
-    async fn test_data(cipher_suite: CipherSuite) -> TestData {
+    async fn test_data(cipher_suite: CipherSuite, generation_auth: GenerationAuthMode) -> TestData {
```

Comment:
Here and elsewhere: why is the variable named `..._auth`, but the type is `...AuthMode`?

Since `GenerationAuthMode` is an enum, it's kind of implicit that it selects a "mode", no? Perhaps it could lose the suffix then. Just a thought, not very critical, but it caught my eye that you're not consistent in the naming :smile:

Reply:
Done, originally named `GenerationAuthMode` to match the sibling `PaddingMode` defined in `EncryptionOptions`.